src/hubs/basehub.ts

```diff
@@ -48,10 +54,11 @@ export class BaseHub extends EventEmitter {
 
     private _type: Consts.HubType;
     private _portMap: {[portName: string]: number} = {};
-    private _attachCallbacks: ((device: Device) => void)[] = [];
+    private _attachCallbacks: ((device: Device) => boolean)[] = [];
 
     constructor (device: IBLEAbstraction, portMap: {[portName: string]: number} = {}, type: Consts.HubType = Consts.HubType.UNKNOWN) {
         super();
+        this.setMaxListeners(20); // Technic Medium Hub has 9 built in devices + 4 external ports. Node.js throws a warning after 11 attached event listeners.
```

Comment:
You can also add a listener for each device type by using `waitForDeviceByType`. There is 23 types + the 13 ports on the Technic Medium Hub and we forgot about the 6 possible virtuals ports.

Lets say somebody is crazy enougth to listen to everything, you could have up to 42 listeners.

Reply:
Sorry my comment is shorthand - the eventListener limit is _per event type_ - in particular the warning is being thrown here is because when the hub detects a new attached device, it creates a new device object, which in turn listens to the "detach" event on the hub to know when to consider itself detached.

I've tested that they're successfully removed, so on a Technic Medium Hub there should never be more than 13 detach events added by this library, plus any extra the user wants (hopefully a single detach event!).

Reply:
Ps. `waitForDeviceByType` doesn't use any event listeners - it merely stashes a callback to be called on an attach, to resolve a promise.

Reply:
I totally brainfarted. Proof I should stop reviewing past midnight...

Reply:
No worries 😄  Thanks for the diligence, it's appreciated! (and necessary!)